logrotate/src/LogRotateHelper.ipp

```diff
@@ -403,7 +450,7 @@ bool LogRotateHelper::createCompressedFile(std::string file_name, std::string gz
         return false;
     }
 
-    int N;
+    size_t N;
     while ((N = fread(buffer, 1, buffer_size, input)) > 0) {
         gzwrite(output, buffer, N);
     }
```

Comment:
@cstamatopoulos  please see the line below also. The result of the if statement below has no impact. Can you in your pull request
 adjust that?
```
if (gzclose(output) != Z_OK);
```